benches/boxed_residue.rs

```diff
@@ -21,6 +21,32 @@ fn bench_montgomery_ops<M: Measurement>(group: &mut BenchmarkGroup<'_, M>) {
         BoxedUint::random(&mut OsRng, UINT_BITS) | BoxedUint::one_with_precision(UINT_BITS),
     )
     .unwrap();
+
+    group.bench_function("invert, U256", |b| {
+        b.iter_batched(
+            || {
+                let modulus = NonZero::new(params.modulus().clone()).unwrap();
+                BoxedResidue::new(BoxedUint::random_mod(&mut OsRng, &modulus), params.clone())
+            },
+            |x| black_box(x).invert(),
+            BatchSize::SmallInput,
+        )
+    });
+
+    group.bench_function("Bernstein-Yang invert, U256", |b| {
+        b.iter_batched(
+            || {
+                let inverter = params.precompute_inverter();
```

Comment:
I think for a fair comparison, this needs to be part of the benchmark loop

Reply:
Well, the idea was to compare the performance once the one-time precomputation penalty had been paid, although it isn't that much, and I'd like to play with replacing the existing implementation of modular inversions for `BoxedResidue` to use Bernstein-Yang for odd moduli, which would benchmark everything.